### PR TITLE
remove annotation code for pipeline run, and use plugin results from

### PR DESCRIPTION
**Pull request by rcerven**
exit task results

* CLOUDBLD-10949

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Changes to metadata also update the documentation for the metadata
- [n/a] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

**Comment by lkolacek** [comment]
Would it make sense to divide the annotation removal part and use plugin results from exit plugin results into separate commits? Even commit message seems to be long to display

File: osbs/tekton.py [hunk 1]
```diff
@@ -663,12 +651,31 @@ def status_status(self):
         return data['status']['conditions'][0]['status']
 
     @property
-    def pipeline_results(self) -> List[Dict[str, str]]:
+    def pipeline_results(self) -> Dict[str, str]:
```
Comment (chmeliik) [inline osbs/tekton.py:654]:
```suggestion
    def pipeline_results(self) -> Dict[str, Any]:
```

File: osbs/tekton.py [hunk 2]
```diff
@@ -494,24 +473,41 @@ def get_error_message(self):
...
+
+            if task_name not in ('binary-container-exit', 'source-container-exit'):
+                continue
+
+            for result in stats['status']['taskResults']:
+                if result['name'] != 'annotations':
+                    continue
+
+                annotations_str = result['value']
+                break
```
Comment (chmeliik) [inline osbs/tekton.py:496]: We have very similar code in get_final_platforms. Could we generalize it? E.g. a `get_task_results` method which returns every result from every taskRun?
https://github.com/containerbuildsystem/osbs-client/blob/e53b2cc3923c7013bc84c77de255bc568c4968a9/osbs/tekton.py#L542

File: osbs/cli/main.py [hunk 3]
```diff
@@ -71,9 +71,8 @@ def _get_build_metadata(pipeline_run, user_warnings_store):
     if pipeline_run.has_succeeded():
         info = pipeline_run.get_info()
         output['pipeline_run']['info'] = info
-        annotations = info['metadata']['annotations']
-        str_repositories = annotations.get('repositories', '{}')
-        all_repositories = json.loads(str_repositories)
+        results = pipeline_run.pipeline_results
+        all_repositories = results.get('repositories', '{}')
```
Comment (chmeliik) [inline osbs/cli/main.py:75]:
```suggestion
        all_repositories = results.get('repositories', {})
```

Comment (rcerven) [reply osbs/cli/main.py:75]: ok

File: osbs/tekton.py (hunk 2 above)
Comment (rcerven) [reply osbs/tekton.py]: done

File: tests/test_api.py [hunk 4]
```diff
@@ -1028,18 +1015,7 @@ def test_get_build_results(self, osbs_binary):
...
-        pipeline_results = [
-            {'name': 'invalid_string', 'value': 'spam'},
-        ]
-        flexmock(PipelineRun).should_receive('pipeline_results').and_return(pipeline_results)
-
-        err_msg = "invalid_string value is not valid JSON: 'spam'"
-
-        with pytest.raises(OsbsValidationException, match=err_msg):
-            osbs_binary.get_build_results('run_name')
+        assert osbs_binary.get_build_results('run_name') == pipeline_results
```
Comment (chmeliik) [inline tests/test_api.py:1018]: Could you update the pipeline_results to something that could be returned by `PipelineRun.pipeline_results`?

File: tests/test_tekton.py [hunk 5]
```diff
@@ -905,28 +982,38 @@ def test_any_task_failed_or_cancelled(
...
                 {"status": {"pipelineResults": [{'name': 'foo', 'value': '1234'}]}},
-                [{'name': 'foo', 'value': '1234'}],
+                {'foo': 1234},
             ),
             (
                 {
                     "status": {
                         "pipelineResults": [
-                            {'name': 'x', 'value': '1'}, {'name': 'y', 'value': '2'},
+                            {'name': 'x', 'value': '{"key": "value"}'}, {'name': 'y', 'value': '2'},
```
Comment (chmeliik) [inline tests/test_tekton.py:995]: Could you add a testcase with `{'name': 'x', 'value': 'null'}`? Those are supposed to be excluded from the returned dict

File: tests/test_api.py (hunk 4 above)
Comment (rcerven) [reply tests/test_api.py:1018]: ok

File: tests/test_tekton.py (hunk 5 above)
Comment (rcerven) [reply tests/test_tekton.py:996]: done

**Review by mkosiarc — APPROVED** [review]

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Review by lkolacek — APPROVED** [review]

**Comment by rcerven** [comment]
rn updated

**Pull request merged**